code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -330,3 +330,59 @@
 
 /obj/item/reagent_containers/glass/beaker/waterbottle/large/empty
 	list_reagents = list()
+
+/obj/item/pestle
+	name = "pestle"
+	desc = "Grind items with a mortar."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "pestle"
+	force = 7
+
+/obj/item/reagent_containers/glass/mortar
+	name = "mortar"
+	desc = "Use this with a pestle to extract chemicals out of items."
+	icon_state = "mortar"
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(5, 10, 15, 20, 25, 30, 50, 100)
+	volume = 100
+	reagent_flags = OPENCONTAINER
+	spillable = TRUE
+	var/obj/item/grinded
+
+/obj/item/reagent_containers/glass/mortar/AltClick(mob/user)
+	if(grinded)
+		grinded.forceMove(drop_location())
+		grinded = null
+		to_chat(user, "You eject the item inside.")
+
+/obj/item/reagent_containers/glass/mortar/attackby(obj/item/I, mob/living/carbon/human/user)
+	if(istype(I,/obj/item/pestle))
+		if(grinded)
+			to_chat(user, "You start grinding...")
+			if((do_after(user, 25, target = src)) && grinded)
+				user.adjustStaminaLoss(40)

```

Comment:
This shouldn't work if the user's stamina is too low. "You're too tired to grind the items!"